File: Day<16>/Solution.py
from collections import deque

class Solution:
    def Reduced_String(self, k, s):
        if k == 1:
            ret = ''
            return ret
        
        q = deque()
        
        for c in s:
            # entries in deque will consist of a char
            # and an integer representing how many times it is repeating
            if len(q) and q[-1][0]==c:
                q[-1][1] += 1           # incrementing count
                if q[-1][1]==k:
                    q.pop()             # popping if count is k
            else:
                q.append( [c,1] )       # adding new character
        
        ret=''
        while len(q):                   # generating output string
            ret += q[0][0] * q[0][1]
            q.popleft()
        
        return ret


#{ 
#  Driver Code Starts
#Initial Template for Python 3

if __name__=="__main__":
    t=int(input())
    for _ in range(t):
        k=int(input())
        s=input().strip()
        obj = Solution()
        print(obj.Reduced_String(k,s))

File: Day<16>/Solution.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;

class Solution{
    public:
    string Reduced_String(int k,string s){
        if(k == 1){
            string ans = "";
            return ans;
        }
        
        stack<pair<char,int>> st;
        string ans="";
        for(int i=0;i<s.size();i++)
        {
            if(st.empty())
            {
                st.push({s[i],1});
            }
            else if(st.top().first==s[i]){
                st.push({s[i],st.top().second+1});
                if(st.top().second==k){
                    int x=k;
                    while(x){
                        st.pop();
                        x--;
                    }
                }
            }
            else if(st.top().first!=s[i]){
                st.push({s[i],1});
            }
        }
        
        while(!st.empty()){
            ans+=st.top().first;
            st.pop();
        }
        reverse(ans.begin(),ans.end());
        return ans;
    }
};
// { Driver Code Starts.

int main() {
    
    
    int t;cin>>t;
    while(t--)
    {
        int k;
        cin>>k;
        string s;
        cin>>s;
        Solution obj;
        cout<<obj.Reduced_String(k,s)<<"\n";
        
    }
    
	return 0;
}
